lib/jobs/ucs-base-job.js

```diff
@@ -0,0 +1,76 @@
+// Copyright 2017 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = ucsBaseJobFactory;
+di.annotate(ucsBaseJobFactory, new di.Provide('Job.Ucs.Base'));
+di.annotate(ucsBaseJobFactory,
+    new di.Inject(
+        'Job.Base',
+        'Logger',
+        'Util',
+        'Promise',
+        'JobUtils.UcsTool',
+        '_',
+        'Assert'));
+
+function ucsBaseJobFactory(
+    BaseJob,
+    Logger,
+    util,
+    Promise,
+    UcsTool,
+    _,
+    assert
+) {
+    function UcsBaseJob(logger, options, context, taskId) {
+        UcsBaseJob.super_.call(this, logger, options, context, taskId);
+
+        var _ucstoolInstance = null;
+        this._getUcsToolInstance = function() {
+            if (!_ucstoolInstance) {
+                _ucstoolInstance = new UcsTool();
+            }
+            return _ucstoolInstance;
+        };
+    }
+    util.inherits(UcsBaseJob, BaseJob);
+
+    UcsBaseJob.prototype._run = function() {};
+
+    UcsBaseJob.prototype._ucsRequest = function(url, settings) {
+        var self = this;
+        var ucsTool = self._getUcsToolInstance();
+        ucsTool.settings = settings;
+        return ucsTool.clientRequest(url);
+    };
+
+    UcsBaseJob.prototype._ucsRequestAsync = function(url, settings, taskId) {
+        var self = this;
+        assert.ok(taskId);
+        return self._ucsRequest(url, settings)
+            .then(function(res) {
+                if (res && res.body && res.body.toUpperCase() !== "ACCEPTED") {
+                    throw new Error(
+                        "Request was not ACCEPTED. Please check input parameters.");
+                }
+                return self._subscribeHttpResponseUuidByPromisify(taskId);
+            });
+    };
+
+    UcsBaseJob.prototype._subscribeHttpResponseUuidByPromisify = function(id) {
```

Comment:
Good job!

Reply:
`_subscribeHttpResponseUuid `is already a `promise`(https://github.com/RackHD/on-tasks/blob/master/lib/jobs/base-job.js#L230), we don't need to promisify it again.

Reply:
OK, since you want to Repromisify `_subscribeHttpResponseUuid`, errors should be processed by calling `callback`.

Reply:
Yes, current code will eat all error scenes. I'm going to add error handler for asynchronous processing, to make sure they can be found and handled.